### PR TITLE
Adds permission sources to User Response

### PR DESCRIPTION
**Pull request by tec-nick**
### Changes

`UsersClient` now returns permissions along with the source of the permissions, according to apparent API specifications. This is done by simply updating the types to pass the permission sources from the http API to the client of this library.

Tests received a minor update to include this change.

### References

This is a response to issue #424.

### Testing

This minor change has only been included on the existing test that calls `UsersClient.GetPermissions`. As I don't believe I have access to the underlying definition of the API, please let me know if there are any differences between this implementation and the underlying API.

- [ ] This change adds unit test coverage

- [X] This change adds integration test coverage

- [X] This change has been tested on the latest version of the platform/language or why not

> This reflects personal testing on the Management API through the portal, and has been tested in my local project, not sure what else I can do.

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

- [ ] All existing and new tests complete without errors

(I can't test this without merging [ref](https://github.com/auth0/auth0.net#testing))

### CODE REVIEW

**Comment by damieng** [comment]
I think it might be better to add PermissionSource directly to Permission rather than introduce UserPermission. 

Changing IPagingList from Permission to UserPermission is a breaking change for anyone that has wrapped/interfaced those classes.

Also what docs did you use for the definition of PermissionSource?

**Comment by tec-nick** [comment]
I'm happy to add `PermissionSource` wherever you prefer. The reason I created a new class was to distinguish it from the permission definition used on `RolesClient.GetPermissionAsync`, (aka `/api/v2/roles/{id}/permissions`). That endpoint does not include this `sources` property.

I could not find a definitive description of the property, I assume the implementation of your http API is not public. In the API helper under [Get a user's permissions](https://auth0.com/docs/api/management/v2#!/Users/get_permissions) (aka `/api/v2/users/{id}/permissions`) it states that there is a property named `sources`, but doesn't list the schema. I tested it with directly applied permissions and with permissions applied via a role to create the schema definition here, but without more information there is the possibility that this definition is incomplete.

**Comment by frederikprijck** [comment]
Thanks for the PR @tec-nick .

I understand the reasoning for introducing `UserPermission` as, as u point out, the permissions returned as part of the roles does not have a `sources` property.

I will check this internally to see if this is expected or not.

File: src/Auth0.ManagementApi/Models/UserPermission.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿namespace Auth0.ManagementApi.Models
+{
+    /// <summary>
+    /// User-specific representation of a user, including permission sources
```
Comment (frederikprijck) [inline src/Auth0.ManagementApi/Models/UserPermission.cs:4]: I think this needs to be User-specific representation of a **permission**.

**Review by frederikprijck — CHANGES_REQUESTED** [review]

**Review by frederikprijck — APPROVED** [review]

**Comment by tec-nick** [comment]
Thanks @frederikprijck for the approval. What are my next steps to getting this branch merged?

**Comment by frederikprijck** [comment]
We had a few internal conversations regarding the breaking change this introduces, hence the delay.

Should be good to go now once CI is succesful

**Pull request merged**